modules/objdetect/test/test_qrcode.cpp

```diff
@@ -43,32 +43,32 @@
 
 namespace opencv_test { namespace {
 
+using std::vector;
+
 TEST(Objdetect_QRCode, regression)
 {
     String root = cvtest::TS::ptr()->get_data_path() + "qrcode/";
-    // String cascades[] =
-    // {
-        // root + "haarcascade_frontalface_alt.xml",
-        // root + "lbpcascade_frontalface.xml",
-        // String()
-    // };
+    String dataset_config = cvtest::TS::ptr()->get_data_path() + "qrcode/dataset_config.json";
+    FileStorage file_config(dataset_config, FileStorage::READ);
```

Comment:
Do we really need this file?
- don't create monster test cases. Use test parameters instead.
- one image - one test. It is too hard to debug what is wrong with 10th picture, because you should process first 9 before that.
- multiple parsing of summary file should be avoided. Configuration per file looks much better (`20110817_030.jpg` + `20110817_030.json`)
- Finally, image name + 4 points can be added into tests code directly.

I hope you collected these validation points in more accurate debug build with all disabled optimizations (like IPP).

Reply:
@alalek @vpisarev  done.